versioned/spi/src/main/java/org/projectnessie/versioned/VersionStore.java

```diff
@@ -110,6 +110,7 @@ default Hash commit(
    * to concurrent readers/writers. The sequence to transplant must be contiguous, in order and
    * share a common ancestor with the target branch.
    *
+   * @param sourceBranch The branch we're transplanting from
```

Comment:
Seems this new parameter's only there to populate the field in `MergeResult`.
I suspect, we can just add 
```java
MergeResult<T> withSourceBranch(BranchName sourceBranch);

    MergeResult<T> withResultType(ResultType resultType);
```
to `MergeResult` and call it from the call sites of `transplant()` and `merge()`. WDYT?

Reply:
_sigh_ fair enough